### PR TITLE
Revert refactor of `pl$DataFrame()` with `schema` only

### PR DESCRIPTION
**Pull request by etiennebacher**
Refactor in #904 considerably slowed down the `DataFrame` creation with `schema` only:

``` r
library(polars)

schema <- list()
for (i in 1:1000) {
  schema[[paste0("var_", i)]] <- pl$Float64
}

bench::mark(
  pre_904 = {
    largs <- lapply(seq_along(schema), \(x) {
      pl$lit(numeric(0))$cast(schema[[x]])$alias(names(schema)[x])
    })
    pl$select(largs)
  }, 
  post_904 = {
    lapply(schema, \(dtype) pl$Series(NULL)$cast(dtype)) |>
      pl$select()
  }
)
#> Warning: Some expressions had a GC in every iteration; so filtering is
#> disabled.
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 pre_904    992.55ms 992.55ms     1.01     10.5MB     17.1
#> 2 post_904      2.99s    2.99s     0.335    14.7MB     10.4
```

### CODE REVIEW

**Review by eitsupi — APPROVED** [review]
The test failure is unrelated this change.

**Comment by eitsupi** [comment]
@etiennebacher I have merged it for now, but have you tested it for other dtypes as well? I mean, this doesn't really cast anything, since the R numeric type is converted to Float64 in Polars, right?

File: R/dataframe__frame.R [hunk 1]
```diff
@@ -279,8 +279,10 @@ pl_DataFrame = function(..., make_names_unique = TRUE, schema = NULL) {
   # no args create empty DataFrame
   if (length(largs) == 0L) {
     if (!is.null(schema)) {
-      out = lapply(schema, \(dtype) pl$Series(NULL)$cast(dtype)) |>
-        pl$select()
+      largs = lapply(seq_along(schema), \(x) {
+        pl$lit(numeric(0))$cast(schema[[x]])$alias(names(schema)[x])
+      })
+      out = pl$select(largs)
```
Comment (eitsupi) [inline R/dataframe__frame.R:285]: Use the pipe operator instead of assigning to `largs`.

**Comment by etiennebacher** [comment]
> have you tested it for other dtypes as well?

Good point, I hadn't done it but the results are the same with a mix of datatypes:

``` r
library(polars)

schema <- list()
for (i in 1:1000) {
  schema[paste0("var_", i)] <- sample(c(pl$Float64, pl$Int16, pl$Binary, pl$String, pl$Categorical()), 1)
}

bench::mark(
  pre_904 = {
    largs <- lapply(seq_along(schema), \(x) {
      pl$lit(numeric(0))$cast(schema[[x]])$alias(names(schema)[x])
    })
    pl$select(largs)
  }, 
  post_904 = {
    lapply(schema, \(dtype) pl$Series(NULL)$cast(dtype)) |>
      pl$select()
  }
)
#> Warning: Some expressions had a GC in every iteration; so filtering is
#> disabled.
#> # A tibble: 2 × 6
#>   expression      min   median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr> <bch:tm> <bch:tm>     <dbl> <bch:byt>    <dbl>
#> 1 pre_904    842.82ms 842.82ms     1.19     7.24MB     15.4
#> 2 post_904      2.47s    2.47s     0.405    12.9MB     10.9
```

**Comment by eitsupi** [comment]
I see, thanks.

**Pull request merged**